drivers/dma/dma_stm32.c

```diff
@@ -102,7 +102,16 @@ static void dma_stm32_irq_handler(const struct device *dev, uint32_t id)
 		stream->busy = false;
 	}
 
-	/* the dma stream id is in range from STM32_DMA_STREAM_OFFSET..<dma-requests> */
+	/* do not continue if dma channel is not enabled */
+#ifdef CONFIG_DMA_STM32_V1
+	if (!LL_DMA_IsEnabledStream(dma, dma_stm32_id_to_stream(id))) {
```

Comment:
A new `stm32_dma_is_enabled_stream()` would be nice.
As we already have `stm32_dma_enable_stream()` and `stm32_dma_disable_stream()`

Reply:
changed: 
checking the channel busy is better because the stm32F4x or stm32F7x (dma  of type V1) could fail in irq endlessly